libbeat/docs/config-file-format.asciidoc

```diff
@@ -0,0 +1,355 @@
+on the current event being processed. Variable expansions are enclosed in
+expansion braces `%{<accessor>:default value}`. Event fields are accessed using
+field references `[fieldname]`. Optional default values can be specified in case the
+field name is missing from the event.
+
+You can also format time stored in the
+`@timestamp` field using the `+FORMAT` syntax where FORMAT is a valid (time
+format)[https://godoc.org/github.com/elastic/beats/libbeat/common/dtfmt].
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+constant-format-string: 'constant string'
+field-format-string: '%{[fieldname]} string'
+format-string-with-date: '%{[fieldname]}-%{+yyyy.MM.dd}'
+------------------------------------------------------------------------------
+
+
+[[config-file-format-env-vars]]
+=== Enironment Variables
+
+Beats support use of environment variables in config files to set values that
+need to be configurable during deployment. Environment variable expansion is
+introduced using `${VAR}`, where `VAR` is the name of the environment variable.
+
+Note: Only values can be set using environment variables. Environment variables
+usage in namespace and setting names are not supported.
+
+Variable references are replaced when settings are read by beats. The
+replacement is case-sensitive and occurs after the YAML file itself has been
+parsed. References to undefined variables will lead to errors when dereferenced
+and no default value is specified. To specify a default value, use:
+
+`${VAR:default_value}`
+
+Where `default_value` is the value to use if the environment variable is
+undefined.
+
+If you need to use a literal `${` in your configuration file then you can write
+`$${` to escape the expansion. The `$` symbol can be used to escape other
+characters in the default_value like using `$}` in order to generate a `}`
+character without closing the variable expansion.
+
+After changing the value of an environment variable, the beat needs to be
+restarted to pick up the new value.
+
+[float]
+=== Examples
+
+Here are some examples of configurations that use environment variables
+and what each configuration looks like after replacement:
+
+[options="header"]
+|==================================
+|Config source	       |Environment setting   |Config after replacement
+|`name: ${NAME}`       |`export NAME=elastic` |`name: elastic`
+|`name: ${NAME:beats}` |no setting            |`name: beats`
+|`name: ${NAME:beats}` |`export NAME=elastic` |`name: elastic`
+|==================================
+
+[[config-file-format-references]]
+=== References
+
+Beats settings can reference other settings splicing multiple optionally custom
+named settings into new values. References use the same syntax as
+<<config-file-format-env-vars>> do. Only fully collapsed setting names can be
+referenced to.
+
+Example referencing `es.host` in `output.elasticsearch.hosts`:
```

Comment:
@urso One more thing to add: Before introducing the concept of custom named settings, I would first introduce a simple example like:
`filebeat.registry_file: ${path.data}/registry`

Maybe even call out specifically that you can reference the path settings.